pkg/queue/health/probe.go

```diff
@@ -103,3 +103,13 @@ func IsHTTPProbeReady(res *http.Response) bool {
 	// response status code between 200-399 indicates success
 	return res.StatusCode >= 200 && res.StatusCode < 400
 }
+
+// IsHTTPProbeShuttingDown checks whether the Response indicates the prober is shutting down.
+func IsHTTPProbeShuttingDown(res *http.Response) bool {
+	if res == nil {
+		return false
+	}
+
+	// status 409 indicates the probe returned a shutdown scenario.
+	return res.StatusCode == http.StatusConflict
```

Comment:
`return res != nil && res.StatusCode == http.StatusConflict`

Reply:
Does it make sense to include check for response body along with status here?

Reply:
actually I don't think this can ever even be nil (["If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close"](https://golang.org/pkg/net/http/#Client.Do)), so we might as well just drop the nil check (though I think we were probably being consistent with the same check in IsHTTPProbeReady so might want to drop there too in that case).

Reply:
Sure, I'm happy to drop the check in both of them in that case. @shashwathi I'm happy to do that; since nobody else seems to be using this, I thought it could be redundant, but I'm happy to add that check.

Reply:
I dropped nil checks in both places now.